ProfileManager.js

```diff
@@ -115,6 +125,82 @@ export default class ProfileManager {
     return await this._getAgentContent({profileAgentRecord});
   }
 
+  /**
+   * Gets the profile agent assigned to the account associated with the
+   * authenticated session for the profile identified by the given profile ID.
+   *
+   * @param {object} options - The options to use.
+   * @param {string} options.profileId - The ID of the profile to get the
+   *   profile agent for.
+   * @param {string} options.capabilityInvoker - The type of invoker to sign
+   * capability invocations.
+   *
+   * @returns {Promise<object>} The profile agent.
+   */
+  async getCapability(
+    {id, profileAgent, profileId, capabilityInvoker = 'local'} = {}) {
+    const capabilityCacheKey = `profileAgent-${profileAgent.id}-zcaps`;
+    const capabilityKey = `${capabilityCacheKey}-${id}-${capabilityInvoker}`;
+
+    if(!this._cache.has(capabilityCacheKey)) {
+      this._cache.set(capabilityCacheKey, new Cache());
+    }
+
+    let promise = this._requests.get(capabilityKey);
+
+    if(promise) {
+      return promise;
+    }
+
+    promise = (async () => {
+      const capabilityCache = this._cache.get(capabilityCacheKey);
+      let zcap = capabilityCache.get(capabilityKey);
+      if(!zcap) {
+        const agentSigner = await this._getInvocationSigner(
+          {profileAgentId: profileAgent.id, capabilityInvoker: 'agent'});
+        if(id === PROFILE_INVOCATION_ZCAP_REFERENCE_ID) {
+          assert.nonEmptyString(profileId, 'profileId');
+          id = _getProfileInvocationZcapKeyReferenceId({
+            zcaps: profileAgent.zcaps,
+            profileId
+          });
+        }
+        const originalZcap = profileAgent.zcaps[id];
+        if(!originalZcap) {
+          const {id: profileAgentId} = profileAgent;
+          throw new Error(
+            `The agent "${profileAgentId}" does not have the zcap: "${id}"`);
+        }
+        if(capabilityInvoker === 'local') {
+          const localSigner = await this._getInvocationSigner(
+            {capabilityInvoker: 'local'});
+          zcap = await utils.delegateCapability({
+            signer: agentSigner,
+            request: {
+              ...originalZcap,
+              parentCapability: originalZcap,
+              controller: localSigner.id
+            }
+          });
+        } else if(capabilityInvoker === 'agent') {
+          zcap = originalZcap;
+        } else {
+          throw new Error('Unsupported invoker for capability.');
```

Comment:
```suggestion
          throw new Error('Unsupported invoker type "${capabilityInvokerType}" for capability.');
```

But, really, we should check validate this at the top of the function and throw a `TypeError` if `local` or `agent` aren't used. Also, those names don't quite cover the difference we want, right? Maybe `ephemeral` instead of `local`? Why did you choose `local`? ... I could potentially be convinced.

EDIT: See above comments ... it seems like this should be an `invocationSignerType` instead where the choices are `local` or `remote`.